scipy/stats/tests/test_multivariate.py

```diff
@@ -132,8 +132,11 @@ def test_degenerate_distributions(self):
 
         def _sample_orthonormal_matrix(n):
             M = np.random.randn(n, n)
-            u, s, v = scipy.linalg.svd(M)
-            return u
+            Q, R = scipy.linalg.qr(M)
```

Comment:
This is not better since both are almost comparable complexity and runs once. In fact stats has a better function https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.ortho_group.html for this usage.

Reply:
On SVD vs QR: I do not believe that `U` needs to be distributed as the Haar measure on `O(N)`. For example, a valid `svd` implementation could produce `U` with positive determinant, or even having biases such as producing `U`'s where all the elements in the first row are positive.

About that function, I noted [in the initial analysis](https://github.com/data-apis/array-api/issues/214#issuecomment-878262647) that it exists, but I figured it would be better not to use it in this test, as we are testing that function in this file.

Reply:
The idea here is to shuffle the matrix entries without changing the basis. Hence the orthonormality is the important bit not the distribution it is sampling from.